Flaky tests: Test_Provider_Registrar_FailsAfterRetries

**Describe the bug**
Provider_Registrar_FailsAfterRetries will occasionally fail on either CI or locally. 

**To Reproduce**
Steps to reproduce the behavior:

While I'm unable to reliably catch the failure, running this a few times will get a failure:
```
go test -timeout 300s github.com/mysteriumnetwork/node/identity/registry -run Test_Provider_Registrar_FailsAfterRetries -count 10000
```

**Expected behavior**
Tests should consistently succeed.

**Additional context**
```
{"level":"info","time":"2020-01-21T09:16:34Z","message":"Stopping provider registrar"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Starting provider registrar"}
 {"level":"debug","time":"2020-01-21T09:16:34Z","message":"Received \"\" service event, ignoring"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Provider \"0x000\" already marked as registered, skipping"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Starting provider registrar"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Provider \"0xsuchIDManyWow\" not registered on BC, will register"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Fees fetched. Registration costs 0"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Registration success for provider \"0xsuchIDManyWow\""}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Stopping provider registrar"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Starting provider registrar"}
 {"level":"error","error":"could not check registration status on BC: explosions everywhere","time":"2020-01-21T09:16:34Z","message":"Could not complete registration for provider \"0xsuchIDManyWow\". Will retry. Retry 0 of 5"}
 {"level":"error","error":"could not check registration status on BC: explosions everywhere","time":"2020-01-21T09:16:34Z","message":"Could not complete registration for provider \"0xsuchIDManyWow\". Will retry. Retry 1 of 5"}
 {"level":"error","error":"could not check registration status on BC: explosions everywhere","time":"2020-01-21T09:16:34Z","message":"Could not complete registration for provider \"0xsuchIDManyWow\". Will retry. Retry 2 of 5"}
 {"level":"error","error":"could not check registration status on BC: explosions everywhere","time":"2020-01-21T09:16:34Z","message":"Could not complete registration for provider \"0xsuchIDManyWow\". Will retry. Retry 3 of 5"}
 {"level":"error","error":"could not check registration status on BC: explosions everywhere","time":"2020-01-21T09:16:34Z","message":"Could not complete registration for provider \"0xsuchIDManyWow\". Will retry. Retry 4 of 5"}
 {"level":"info","time":"2020-01-21T09:16:34Z","message":"Stopping provider registrar"}
 --- FAIL: Test_Provider_Registrar_FailsAfterRetries (0.00s)
     provider_registrar_test.go:119: 
         	Error Trace:	provider_registrar_test.go:119
         	            				asm_amd64.s:1357
         	Error:      	Expected value not to be nil.
         	Test:       	Test_Provider_Registrar_FailsAfterRetries
 FAIL
 coverage: 22.4% of statements
 FAIL	github.com/mysteriumnetwork/node/identity/registry	0.142s
```

--- a/identity/registry/provider_registrar_test.go
+++ b/identity/registry/provider_registrar_test.go
@@ -19,7 +19,6 @@ package registry
 
 import (
 	"testing"
-	"time"
 
 	"github.com/mysteriumnetwork/node/core/service"
 	"github.com/mysteriumnetwork/node/identity"
@@ -121,8 +120,6 @@ func Test_Provider_Registrar_FailsAfterRetries(t *testing.T) {
 	}()
 
 	registrar.consumeServiceEvent(mockEvent.event)
-	time.Sleep(time.Millisecond * 2)
-	registrar.stop()
 	<-done
 }
 

